tests/test_encryption.py

```diff
@@ -311,20 +312,46 @@ def test_decrypt_token_v2_custom_now(self):
         keys = EncryptionKeysCollection([_master_key, _site_key])
 
         with self.assertRaises(EncryptionError):
-            result = decrypt_token(token, keys, now=expiry+dt.timedelta(seconds=1))
+            result = decrypt(token, keys, now=expiry + dt.timedelta(seconds=1))
 
-        result = decrypt_token(token, keys, now=expiry-dt.timedelta(seconds=1))
+        result = decrypt(token, keys, now=expiry - dt.timedelta(seconds=1))
         self.assertEqual(_example_id, result.uid2)
 
 
+    def test_smoke_token_v3(self):
+        uid2 = _example_id
+        identity_scope = IdentityScope.UID2
+        now = dt.datetime.now(tz=timezone.utc)
+
+        keys = EncryptionKeysCollection([_master_key, _site_key, _keyset_key], default_keyset_id=20,
+                                        master_keyset_id=9999, caller_site_id=20)
+
+        result = encrypt(uid2, identity_scope, keys, now=now, ad_token_version=AdvertisingTokenVersion.ADVERTISING_TOKEN_V3)
+        final = decrypt(result, keys, now=now)
+
+        self.assertEqual(uid2, final.uid2)
+
+    def test_smoke_token_v4(self):
+        uid2 = _example_id
+        identity_scope = IdentityScope.UID2
+        now = dt.datetime.now(tz=timezone.utc)
+
+        keys = EncryptionKeysCollection([_master_key, _site_key, _keyset_key], default_keyset_id=20,
```

Comment:
similar question - the keyset_id/master_keyset_id looks different to java sdk's equivalent unit tests so wondering if we should make it consistent @jon8787 ?

Reply:
I can't find the java equivalent for this one? I made this one up myself

Reply:
right i thought you are adding corresponding methods to 
https://github.com/IABTechLab/uid2-client-java/blob/master/src/test/java/com/uid2/client/EncryptionTestsV4.java#L108
and equivalent's in com.uid2.client.EncryptionTestsV3#smokeTest

Reply:
oh, yeah ideally the input for this test should be JSON rather than EncryptionKeysCollection